src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  sourceString
+    .split(';')
+    .map((item) => item.trim())
+    .filter(Boolean)
+    .forEach((line) => {
+      const [key, value] = line.split(':').map((item) => {
+        const trimValue = item.trim();
+        const removeSemiColumn = trimValue.split(';').join('');
+        const removeLineBreakValue = removeSemiColumn.replace(/\r/g, '');
+
+        return removeLineBreakValue.trimEnd();
```

Comment:
Using `trimEnd()` here is redundant since `trim()` on line 17 already removes whitespace from both ends of the string. Consider removing this line to avoid unnecessary operations.